settings.lua

```diff
@@ -3275,15 +3304,18 @@ end
 
 ---@class BunkerSpeedSetting : SpeedSetting
 BunkerSpeedSetting = CpObject(SpeedSetting)
-function BunkerSpeedSetting:init(vehicle)
-	SpeedSetting.init(self, 'bunkerSpeed','COURSEPLAY_MODE10_MAX_BUNKERSPEED', 'COURSEPLAY_MODE10_MAX_BUNKERSPEED', vehicle,5,20) 
+---@param vehicle table
+---@param levelCompactModeSetting LevelCompactModeSetting
+function BunkerSpeedSetting:init(vehicle, levelCompactModeSetting)
+	self.levelCompactModeSetting = levelCompactModeSetting
+	SpeedSetting.init(self, 'bunkerSpeed','COURSEPLAY_MODE10_MAX_BUNKERSPEED', 'COURSEPLAY_MODE10_MAX_BUNKERSPEED', vehicle,5,20)
 	self:set(10)
 	self.MAX_SPEED_LEVELING = 10
 end
 
 function BunkerSpeedSetting:checkAndSetValidValue(x)
 	local new = SettingList.checkAndSetValidValue(self, x)
-	if self.vehicle.cp.settings.levelCompactMode:hasLeveler() then
```

Comment:
@schwiti6190 Just FYI, I refactored this as it was causing a circular dependency between this setting and the global cp.settings variable. We should avoid referencing global variables from a class where possible.

Reply:
This is a great idea, will try to do the same for different settings.